src/impression.js

```diff
@@ -235,3 +241,65 @@ function applyResponse(win, response) {
     win.history.replaceState(null, '', newHref);
   }
 }
+
+/**
+ * Return a promise that whether appending extra url params to outgoing link is required.
+ * @param {!./service/ampdoc-impl.AmpDoc} ampdoc
+ * @return {!Promise<boolean>}
+ */
+export function shouldAppendExtraParams(ampdoc) {
+  return whenDocumentReady(ampdoc.win.document).then(doc => {
```

Comment:
It should be `ampdoc.whenReady()` or such. Sorry, should have been more specific.